File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel

from app.models.document import DocumentStatus


class DocumentUploadResponse(BaseModel):
    id: int
    knowledge_base_id: int
    filename: str
    file_type: str
    file_md5: str
    status: DocumentStatus
    task_id: str | None
    created_at: datetime


class DocumentChunkPreview(BaseModel):
    id: int
    chunk_index: int
    content: str
    source_page: int | None
    source_file: str
    created_at: datetime


class DocumentOperationLogResponse(BaseModel):
    id: int
    operation_type: str
    status: str
    task_id: str | None
    stage: str | None
    elapsed_ms: int | None
    message: str | None
    created_at: datetime


class DocumentDetailResponse(DocumentUploadResponse):
    chunk_count: int
    error_message: str | None
    page_count: int
    extracted_preview: str | None
    processing_started_at: datetime | None
    processing_finished_at: datetime | None
    retry_count: int
    processing_duration_ms: int | None
    delete_task_id: str | None


class DocumentProgressMessage(BaseModel):
    document_id: int
    task_id: str | None = None
    status: DocumentStatus
    progress: int
    message: str


class DocumentListQuery(BaseModel):
    kb_id: int
    status: DocumentStatus | None = None

File: frontend/src/types/document.ts
export type DocumentStatus = 'pending' | 'running' | 'completed' | 'failed' | 'deleting' | 'delete_failed' | 'deleted'

export interface DocumentItem {
  id: number
  knowledge_base_id: number
  filename: string
  file_type: string
  file_md5: string
  status: DocumentStatus
  task_id: string | null
  delete_task_id: string | null
  chunk_count: number
  error_message: string | null
  created_at: string
  page_count: number
  extracted_preview: string | null
  processing_started_at: string | null
  processing_finished_at: string | null
  retry_count: number
  processing_duration_ms: number | null
}

export interface DocumentChunkPreview {
  id: number
  chunk_index: number
  content: string
  source_page: number | null
  source_file: string
  created_at: string
}

export interface DocumentOperationLog {
  id: number
  operation_type: string
  status: string
  task_id: string | null
  stage: string | null
  elapsed_ms: number | null
  message: string | null
  created_at: string
}

export interface DocumentProgressMessage {
  document_id: number
  task_id: string | null
  status: DocumentStatus
  progress: number
  message: string
  stage?: string
}
